fastapi/routing.py

```diff
@@ -222,7 +227,25 @@ async def app(request: Request) -> Response:
         )
         values, errors, background_tasks, sub_response, _ = solved_result
         if errors:
-            raise RequestValidationError(errors, body=body)
+            has_dict_error = False
+            content_type = request.headers.get("content-type")
+            is_multipart_body = "multipart" in content_type if content_type else False
+            for e in errors:
+                if isinstance(e, list):
+                    for err in e:
+                        if isinstance(err.exc, DictError):
+                            has_dict_error = True
+                else:
+                    if isinstance(e.exc, DictError):
+                        has_dict_error = True
+
+            if is_multipart_body and has_dict_error:
+                raise RequestInvalidContentTypeException(
+                    status_code=415,
+                    detail="Unsupported media type: multipart/form-data (expected application/json)",
+                )
+            else:
```

Comment:
Remove else statement, no need for that.

Reply:
I put the else statement because I want to keep some scenario where the 422 status code response will still be needed.